sqlx-core/src/decode.rs

```diff
@@ -77,3 +81,33 @@ where
         }
     }
 }
+
+macro_rules! impl_decode_for_smartpointer {
+    ($smart_pointer:ty) => {
+        impl<'r, DB, T> Decode<'r, DB> for $smart_pointer
+        where
+            DB: Database,
+            T: Decode<'r, DB>,
```

Comment:
I believe this may actually _remove_ the impls for `Box<str>` and `Box<[u8]>` because of the implicit `T: Sized` bound here.